hilti/include/ast/scope-lookup.h

```diff
@@ -34,8 +34,7 @@ extern std::pair<bool, Result<std::pair<NodeRef, ID>>> lookupID(const ID& id, co
  */
 template<typename D>
 Result<std::pair<NodeRef, ID>> lookupID(const ID& id, const visitor::Position<Node&>& p) {
-    auto i = p.path.rbegin();
-    while ( i != p.path.rend() ) {
+    for ( auto i = p.path.rbegin(); i != p.path.rend(); i++ ) {
```

Comment:
While this is a likely cheap `std::vector` iterator, we have iterators in hilti-rt which are not that cheap to construct, so the preferred pattern if possible should always be precrement.
```suggestion
    for ( auto i = p.path.rbegin(); i != p.path.rend(); ++i ) {
```